src/Chat/utils/validators.js

```diff
@@ -54,15 +57,18 @@ const chatOptions = maybeOptions => {
 const clientOptions = maybeOptions => {
   const shape = {
     username: isString,
-    oauth: isString,
+    password: isString,
```

Comment:
My fear with changing this from `oauth` to `password` is that users will try their actual _Twitch password_ instead of the required OAuth token. If you think `oauth` is unclear, would `token` be better?

Reply:
The `password` property is a computed one based on `token`. I chose this one because the PASS can be the token with `oauth:` or the guest one without it.

Reply:
To connect to Twitch IRC, you supply an OAuth token, not a password; this property should not be called _password_.

Reply:
You can supply an OAuth token prefixed by `oauth:` to tell the server it's one **OR** a random string used during anonymous login (SCHMOOPIIE, TWITCHJS, LOREMIPSUM, etc).

I don't see why I should use another word than `password` and like I said before, it's only a computed property which uses `token` in the first place (https://github.com/twitch-apis/twitch-js/blob/68f2630d723a59c280cc1e7d2c39cb2ce6e08e3a/src/Chat/utils/validators.js#L70).

If you don't like this way, it's also possible to sanitize the token option when the PASS is sent to the server (https://github.com/twitch-apis/twitch-js/blob/52cf6f7b75fbf354e6eb58c3c842276f43694e49/src/Chat/Client.js#L76).

Reply:
OK I see where you are coming from. `Client` options are shielded from the end user, so the name of this property does not really matter. Additionally, `password` makes sense because of the `PASS` command.

What about [the `token` property name change](https://github.com/twitch-apis/twitch-js/pull/163/files#diff-127f3397ec5ba7113b1cc4c6eb7dcc2cR28) in `Chat` options though? This not only will confuse end users, but is also a breaking change to current users of v2 beta.

Reply:
I don't see why it would confuse people because of an additional property, `token` will still be there.
And I agree, `password` should be computed on `Client` only, I'll change that.

Reply:
@Seldszar,

Sorry, this review is taking so long. I really appreciate this discussion and the hard work you have put into this PR. In a day, I may look at several very different code bases, so it takes me a while to adjust to even my own code. Plus, I have not had much time to devote to `twitch-js` recently, so I really appreciate your patience ❤️.

Even though, in the end, the end user supplies `{ ..., token: <TOKEN>, ... }` to [Chat options](https://github.com/twitch-apis/twitch-js/blob/52cf6f7b75fbf354e6eb58c3c842276f43694e49/src/Chat/utils/validators.js#L39), and the library sanitizes it and handles it internally (no matter what it is called), I think this highlights a mistake in architecture I made; that is, changing the variable name where the token is stored from `token` to `oauth`, (and now to `password` with this PR).

This has been quite confusing for me, and it is my own fault, but if I am confused, I imagine it will be even more confusing for developers contributing to this codebase. For instance, I imagine that as a developer, if I supply a library with `{ ..., token: <TOKEN>, ... }`, and a bug related to the token arises, I would hope that I could easily trace the token through the library just by the variable name alone, and resolve the bug. Instead, I now need knowledge of the `Chat` options validator and password sanitizer functions. Changing the variable name from `oauth` or `password` **obscures the implementation**, in my opinion, for no good reason. The JSDoc also does not make sense. It describes the shape, but `shape = { ... }` does not include a `token` property.

No one has time to maintain open source libraries, so I think it is important to reduce, as much as possible, the cognitive load required to maintain `twitch-js`. This means striving for simple, predictable and accessible code. You have already done this with the sanitizer functions 🥇; making the token variable name consistent across the code base would go even further.

Please either:
1. revert these changes to this variable name, or
1. make this variable name more consistent by changing all instances to `token`

... and I will approve this PR.

Thank you for understanding ❤️

(If you choose to revert, I will make a follow-up PR to make the variable name consistent.)

Reply:
I understand. I think I'll leave it to you if you know how to do this.